dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -548,17 +697,131 @@ public void test_MultilangContent_IndexingAllLanguages()
+                    Assert.assertEquals(pageSecondLang.getLanguageId(),
+                            searchResults.getResults().get(0).getLanguage());
+                }
+
+                if(!testCase.expectedResultsWhenSearchingContentInSecondLanguage.pageInDefaultLang
+                        && !testCase.expectedResultsWhenSearchingContentInSecondLanguage.pageInSecondLang) {
+                    Assert.assertEquals("Content in second Language gives NO results", 0,
+                            searchResults.getTotalResults());
+                }
+            }
 
         } finally {
             Config.setProperty("DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE",
                     defaultContentToDefaultLangOriginalValue);
+
+            Config.setProperty("DEFAULT_PAGE_TO_DEFAULT_LANGUAGE",
+                    defaultPagetoDefaultLangOriginalValue);
         }
     }
 
+    private HTMLPageAsset createNewVersionHtmlPage(HTMLPageAsset pageDefaultLang, Language secondLang)
+            throws DotSecurityException, DotDataException {
+        Contentlet pageSecondLang = contentletAPI.find(pageDefaultLang.getInode(), systemUser, false);
+        pageSecondLang.setInode("");
+        pageSecondLang.setLanguageId(secondLang.getId());
+        pageSecondLang = contentletAPI.checkin(pageSecondLang, systemUser, false);
+        ContentletDataGen.publish(pageSecondLang);
+        return APILocator.getHTMLPageAssetAPI().fromContentlet(pageSecondLang);
+    }
+
+    private void createNewVersionAndPublishExistingEmployeeContent(Language secondLang,
+            Contentlet contentletDefaultLang, String firstName) throws DotDataException, DotSecurityException {
+        Contentlet contentletSecondLang = contentletAPI
+                .find(contentletDefaultLang.getInode(), systemUser, false);
+        contentletSecondLang.setInode("");
+        contentletSecondLang.setLanguageId(secondLang.getId());
+        contentletSecondLang.setStringProperty("firstName", firstName);
+        contentletSecondLang = contentletAPI.checkin(contentletSecondLang, systemUser, false);
+        ContentletDataGen.publish(contentletSecondLang);
+    }
+
+    private Contentlet createAndPublishEmployeeContent(Host site, Language language,
+            String firstName) throws DotDataException, DotSecurityException {
+        Contentlet contentletDefaultLang = TestDataUtils
+                .getEmployeeContent(true, language.getId(), null, site);
+
+        contentletDefaultLang = contentletAPI
+                .find(contentletDefaultLang.getInode(), systemUser, false);
+        contentletDefaultLang.setStringProperty("firstName", firstName);
+        contentletDefaultLang = contentletAPI.checkin(contentletDefaultLang, systemUser, false);
+        ContentletDataGen.publish(contentletDefaultLang);
+        return contentletDefaultLang;
+    }
+
+    private HTMLPageAsset createHtmlPageAsset(Language lang1, Contentlet contentlet)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'lang1' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)